tests/load/pipeline/test_pydantic_models.py

```diff
@@ -0,0 +1,41 @@
+import dlt
+import typing as t
+
+from pydantic import BaseModel
+from dlt.common.libs.pydantic import DltConfig
+
+
+def test_flattened_model() -> None:
+    class Child(BaseModel):
+        child_attribute: str
+        optional_child_attribute: t.Optional[str] = None
+
+    class Parent(BaseModel):
+        child: Child
+        optional_parent_attribute: t.Optional[str] = None
+        dlt_config: t.ClassVar[DltConfig] = {"skip_complex_types": True}
+
+    example_data = {
+        "optional_parent_attribute": None,
+        "child": {"child_attribute": "any string", "optional_child_attribute": None},
+    }
+
+    @dlt.resource
```

Comment:
Re: my comment above, I think we should test for both cases (i.e. `skip_complex_types = {True/False}`)
```python
# Test Case 1: "skip_complex_types" = False -> both pydantic models and dicts should be coerced into complex column
class Child(BaseModel):
    child_attribute: str
    optional_child_attribute: t.Optional[str] = None


class Parent(BaseModel):
    optional_parent_attribute: t.Optional[str] = None
    child: Child
    dct: t.Dict[str, t.Any]
    dlt_config: t.ClassVar[DltConfig] = {"skip_complex_types": False}


example_data = {
    "optional_parent_attribute": None,
    "dct": {"child_attribute": "any string", "optional_child_attribute": None},
}

expected_loaded_data = {
    "child__child_attribute": "any string",
    "dct": '{"child_attribute": "any string", "optional_child_attribute": None}',
}


# Test Case 2 "skip_complex_types" = False -> both pydantic models and dicts should be flattened. Optional fields should be kept only for pydantic models
class Child(BaseModel):
    child_attribute: str
    optional_child_attribute: t.Optional[str] = None


class Parent(BaseModel):
    optional_parent_attribute: t.Optional[str] = None
    child: Child
    dct: t.Dict[str, t.Any]
    dlt_config: t.ClassVar[DltConfig] = {"skip_complex_types": True}


example_data = {
    "optional_parent_attribute": None,
    "dct__child_attribute": "any string",
    "child__child_attribute": "any string", 
    "child__optional_child_attribute": None,
}

expected_loaded_data = {"optional_parent_attribute": None, "dct__child_attribute": "any string"}

```